src/ess/sans/sans.py

```diff
@@ -0,0 +1,107 @@
+# SANS specific functions
+import scipp as sc
+import scippneutron as scn
+from .contrib import map_to_bins, to_bin_centers
+from .reduction import reduce_to_q, simple_reducer, grouping_reducer
+from .normalization import solid_angle, transmission_fraction
+
+
+def to_wavelength(
+    data,
+    transmission,
+    direct_beam,
+    direct,
+    masks,
+    wavelength_bins,
+    tof_bins_monitors,
+    pixel_size,
+    pixel_length,
+):
+    """
+    TOF to wavelength conversion
+    """
+    data = data.copy()
```

Comment:
The only modifying change to `data` below is setting masks, so a shallow copy should be enough here (and faster):
```suggestion
    data = data.copy(deep=False)
```